✨ Feat      : ChatPage 헤더부분 컴포넌트로 변경, List 컴포넌트 조건부스타일렌더링 props로 변경

Meow-Chu opened the pull request:
1. ChatPage내 코드로 작성한 헤더 -> 헤더컴포넌트(components/common/header.jsx)로 교체
2. ChatList의 List 컴포넌트부분 변경사항 : 클래스네임으로 설정했던 조건부스타일렌더링을 props로 바꾸었음

> ### 잠깐! PR하기 전에 확인해주세요!

✅ PR 제목의 형식을 잘 작성했나요?(커밋메시지 컨벤션과 동일)   
✅ 불필요한 코드를 제거했나요?   
✅ 작업 시작 전 후로 develop브랜치가 최신상태임을 확인했나요?   
✅ 라벨은 등록했나요?   
<br>

✨ PR 한 줄 요약
-------------------------------------------------
ChatPage 헤더부분 컴포넌트로 변경, List 컴포넌트 조건부스타일렌더링 props로 변경
<br>
<br>

🛠 상세 작업 내용
-------------------------------------------------  
- 채팅리스트 : 헤더코드를 공통컴포넌트헤더로 바꾸기
-  채팅리스트 : 스타일드컴포넌트에 클래스네임 썼던 부분 프롭스로 바꾸기
<br>
<br>

📸 참고 사항
-------------------------------------------------
![image](https://github.com/FRONTENDSCHOOL7/final-05-oguogu/assets/129969049/4f997ace-3e75-4adf-9bba-d832ca2911c2)

![image](https://github.com/FRONTENDSCHOOL7/final-05-oguogu/assets/129969049/99ed26dd-a61a-4488-8f44-8af0373807ed)


<br>
<br>


💡 관련 이슈
-------------------------------------------------
[Feature] 채팅목록페이지 구현 #2 
[Feature] 공통컴포넌트(버튼,헤더) 작업 #12 
<br>
<br>

Meow-Chu committed:
✨ Feat      : ChatPage내 코드로 작성한 헤더 -> 헤더컴포넌트(components/common/heade.jsx)로 교체
 ChatList의 List 컴포넌트부분 변경사항 : 클래스네임으로 설정했던 조건부스타일렌더링을 props로 바꾸었음

yujin1228 commented on src/pages/ChatPage/ChatPage.style.jsx:
```diff
@@ -27,9 +19,7 @@ export const List = styled.li`
-  &.selected {
-    background-color: var(--main-color-02);
-  }
+  background-color: ${(props) => (props.selectedItem === props.id ? 'var(--main-color-02)' : '')};
```
이 부분은 props.selectedItem === props.id 의 결과가 true일때만 색상을 주는거니까 
삼항연산자 대신 (props.selectedItem === props.id && 'var(--main-color-02)' )  이렇게 쓰는 방법도 있습니다!

Meow-Chu committed:
✨ Feat      : ChatList 를 클릭했을 때 배경색을 변하게 하는 코드 수정

삼항연산자였던 코드를 아래와 같이 수정하였습니다
background-color: ${(props) => props.selectedItem === props.id && 'var(--main-color-02)'};
